types/src/main/scala/aqua/types/Type.scala

```diff
@@ -100,21 +184,23 @@ case class OptionType(element: Type) extends BoxType {
   override def toString: String = "?" + element
 }
 
-case class ProductType(name: String, fields: NonEmptyMap[String, Type]) extends DataType {
+// Struct is an unordered collection of labelled types
+case class StructType(name: String, fields: NonEmptyMap[String, Type]) extends DataType {
 
   override def toString: String =
     s"$name{${fields.map(_.toString).toNel.toList.map(kv => kv._1 + ": " + kv._2).mkString(", ")}}"
 }
 
-case class ArrowType(args: List[Type], res: Option[Type]) extends Type {
+case class ArrowType(domain: ProductType, codomain: ProductType) extends Type {
```

Comment:
docs
and i don't think that it is a good idea to rename arguments to `domain` and `codomain`